sycl/include/CL/sycl/detail/accessor_impl.hpp

```diff
@@ -141,6 +141,23 @@ class LocalAccessorImplHost {
   int MDims;
   int MElemSize;
   std::vector<char> MMem;
+
+  bool PerWI = false;
+  size_t LocalMemSize;
+  size_t MaxWGSize;
+  void resize(size_t LocalSize, size_t GlobalSize) {
+    if (GlobalSize != 1 && LocalSize != 1) {
+      // If local size is not specified then work group size is chosen by
+      // runtime. That is why try to allocate based on max work group size or
+      // global size. In the worst case allocate 80% of local memory.
```

Comment:
Because I think it is not good to allocate 100% of local mem, then it will not be possible to use local mem for other purposes inside the kernel. If [# work items running in parallel] * [flush buffer size provided by user] exceeds 80% of local memory then data for work items that don't have flush buffer will not be printed. But as far as I know  maximum number of work items in work group for each device is defined, so this number could not be too big. Theoretically user can provide any flush buffer size, but I am not sure that any implementation can handle any input provided by user. Probably we can write an error message if user provided too big flush buffer size so that allocated local memory is not enough.

This design was discussed with @rolandschulz and @dbabokin.
@rolandschulz @dbabokin Could you please also provide your input regarding this question?

Reply:
Can you defer this computation somehow till after the run-time has chosen the wg-size? Or do we always choose max WG-size/GlobalSize so this matches?

We should throw an exception if the flush buffer is too big.